sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -738,6 +738,10 @@ class Analyzer(
             if findAliases(aggregateExpressions).intersect(conflictingAttributes).nonEmpty =>
           (oldVersion, oldVersion.copy(aggregateExpressions = newAliases(aggregateExpressions)))
 
+        case oldVersion @ FlatMapGroupsInPandas(_, _, output, _)
+            if AttributeSet(output).intersect(conflictingAttributes).nonEmpty =>
```

Comment:
Why not using `oldVersion.outputSet`?

Reply:
cc @maryannxue Deduplicating on conflicting attributes in this function is easily broken. In the long term, this is not the perfect way to handle it. We should consider to fundamentally fix it.

Reply:
@gatorsmile I agree with you. Moreover, there are other possible problems in having the same expressions (with same exprId) in different part of a tree (please see SPARK-24051). So probably on long term we can add a specific rule for addressing this problem (extending/generalizing what I tried to do in SPARK-24051). What do you think?